setup.py

```diff
@@ -53,22 +53,26 @@ def read(*rnames):
     extras_require=dict(
         test=[
             "plone.app.testing[robot]>=4.2.2",
-            "plone.app.robotframework",
-            "plone.dexterity",
-            "Products.CMFCore",
             "requests",
+            "BTrees",
+            "plone.app.contenttypes[test]",
+            "plone.app.layout",
+            "plone.app.redirector",
+            "plone.app.textfield",
+            "plone.namedfile",
+            "plone.testing",
+            "z3c.relationfield",
+            "zope.intid",
+            "plone.base",
         ]
     ),
     install_requires=[
         "setuptools",
         "collective.monkeypatcher",
-        "zope.component",
```

Comment:
@gforcada do I understand correctly that we are not explicit about the libraries that we use in install_requires any longer? Any dependency of Products.CMFCore will not be listed here in the future, right? I don't have a strong opinion here. I'd just like to understand our current best practices, so I can stick to them.

Reply:
@gforcada interesting. I didn't know about this problem with the delay in installation. So this is yet another reason to use the Plone package dependencies.

My suggestion for improvement is also what I said in the previous comment, to try to do this automatically. Dependencies always change.

We also have the problem of packages that are compatible with Plone 6.0 and 6.1. If a package is no longer a dependency on Plone 6.1, but remains a dependency on Plone 6.0, it should be in the install_requires of a package that is compatible with Plone 6.0 and 6.1. So the list of ignored packages should contain all dependencies of both versions of Plone. And in the future include Plones 6.2, 7.0... This is difficult to maintain. So if there is a way for this to be automatic it would be wonderful.

Reply:
`pip` being a bit slower to install, is not so much important, specially in the day of getting things CI/CD that you don't have to wait on your local computer to do something else afterwards. It is still noticeable, but not too big of a deal ⚖️ 

Getting `z3c.dependencychecker` (or any other dependency checker for that matter) smarter would be great indeed, ideas/PRs welcome 😄 https://github.com/reinout/z3c.dependencychecker/

Handle 6.0/1 dependencies, and even 7.0 at some point... that's hard... something that could help a bit would be https://github.com/reinout/z3c.dependencychecker/issues/106 i.e. try to be smart when detecting a new dependency and see if it is wrapped in a `try/except` block and thus do not consider it a hard dependency...

But that's hairy to get a specific answer for all cases.

Reply:
@gforcada thanks for elaborating on this. I think it makes sense to automate those kind of things as much as possible. Automation and consistency beat 100% accuracy.